differ/differ.go

```diff
@@ -294,7 +302,183 @@ func evaluateFilterExpression(eventFilter string, thresholds EventThresholds, ev
+			Str("cluster name", string(cluster.ClusterName)).
+			Msg("Rendering reports failed for this cluster")
+		return
+	}
+	for _, r := range reports {
+		module := r.Module
+		ruleName := moduleToRuleName(module)
+		errorKey := r.ErrorKey
+
+		likelihood, impact, totalRisk, _ := findRuleByNameAndErrorKey(ruleContent, ruleName, errorKey)
+		eventValue := EventValue{
+			Likelihood: likelihood,
+			Impact:     impact,
+			TotalRisk:  totalRisk,
+		}
+
+		// try to evaluate event filter expression
+		result, err := evaluateFilterExpression(serviceLogEventFilter,
+			serviceLogEventThresholds, eventValue)
+		if err != nil {
+			log.Err(err).Msg(evaluationErrorMessage)
+			continue
+		}
+
+		if result > 0 {
+			renderedReport, err := findRenderedReport(renderedReports, ruleName, errorKey)
+			if err != nil {
+				log.Err(err).Msgf("Output from content template renderer does not contain "+
+					"result for cluster %s, rule %s and error key %s", cluster.ClusterName, ruleName, errorKey)
+				continue
+			}
+
+			logEntry := types.ServiceLogEntry{
+				ClusterUUID: string(cluster.ClusterName),
+				Description: renderedReport.Description,
+				ServiceName: "CCX Notification Service",
+				Summary:     renderedReport.Reason,
+			}
+
+			// It is necessary to truncate the fields because of Service Log limitations
+			if len(logEntry.Description) > 255 {
+				logEntry.Description = logEntry.Description[:255]
+			}
+
+			if len(logEntry.Summary) > 255 {
+				logEntry.Summary = logEntry.Summary[:255]
+			}
+
+			msgBytes, err := json.Marshal(logEntry)
+			if err != nil {
+				log.Error().Err(err).Msg(invalidJSONContent)
+				continue
+			}
+
+			_, _, err = serviceLogNotifier.ProduceMessage(msgBytes)
+			if err != nil {
+				log.Err(err).
+					Str("cluster", string(cluster.ClusterName)).
+					Str("rule", string(ruleName)).
+					Str("error key", string(errorKey)).
+					Msgf("Sending entry to service log failed for this report")
+				continue
+			}
+		}
+	}
+}
+
+func produceEntriesToKafka(cluster types.ClusterEntry, ruleContent types.RulesMap,
+	reportItems []types.ReportItem, storage Storage, report types.ClusterReport) int {
+
+	notificationMsg := generateInstantNotificationMessage(
+		&notificationClusterDetailsURI,
+		fmt.Sprint(cluster.AccountNumber),
+		fmt.Sprint(cluster.OrgID),
+		string(cluster.ClusterName))
+	notifiedAt := types.Timestamp(time.Now())
+
+	for _, r := range reportItems {
+		module := r.Module
+		ruleName := moduleToRuleName(module)
+		errorKey := r.ErrorKey
+		likelihood, impact, totalRisk, description := findRuleByNameAndErrorKey(ruleContent, ruleName, errorKey)
+		eventValue := EventValue{
+			Likelihood: likelihood,
+			Impact:     impact,
+			TotalRisk:  totalRisk,
+		}
+
+		// try to evaluate event filter expression
+		result, err := evaluateFilterExpression(kafkaEventFilter,
+			kafkaEventThresholds, eventValue)
+
+		if err != nil {
+			log.Err(err).Msg(evaluationErrorMessage)
+			continue
+		}
+
+		if result > 0 {
```

Comment:
later we might need to think about how to refactor these two exact the same codes :)